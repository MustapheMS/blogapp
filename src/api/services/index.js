import users from './users/users.service';
import posts from './posts/posts.service';

export default (app) => {
  app.configure(users);
  app.configure(posts);
};
