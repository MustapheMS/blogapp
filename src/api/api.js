import compress from 'compression';
import helmet from 'helmet';
import cors from 'cors';
import feathers from '@feathersjs/feathers';
import configuration from '@feathersjs/configuration';
import express from '@feathersjs/express';
import socketio from '@feathersjs/socketio';

import logger from './logger';
import middleware from './middleware';
import services from './services';
import apiHooks from './api.hooks';
import channels from './channels';

import authentication from './authentication';

import mongoose from './mongoose';

const api = express(feathers());
api.use(helmet());
api.use(cors());
api.use(compress());

// Load app configuration
api.configure(configuration());
api.use(express.json());
api.use(express.urlencoded({ extended: true }));

// Set up Plugins and providers
api.configure(express.rest());
api.configure(socketio());

api.configure(mongoose);

// Configure other middleware (see `middleware/index.js`)
api.configure(middleware);
api.configure(authentication);
// Set up our services (see `services/index.js`)
api.configure(services);
// Set up event channels (see channels.js)
api.configure(channels);

// Configure a middleware for 404s and the error handler
api.use(express.notFound());
api.use(express.errorHandler({ logger }));

api.hooks(apiHooks);

export default api;
