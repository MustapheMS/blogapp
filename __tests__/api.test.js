import ms from 'ms';
import delay from 'delay';
import fetch from 'node-fetch';
import url from 'url';
import api from '../src/api/api';
import userData from './__fixtures__/user';

const port = api.get('port');
let server;

const getUrl = pathname => url.format({
  hostname: api.get('host') || 'localhost',
  protocol: 'http',
  port,
  pathname,
});

beforeAll(async () => {
  await new Promise((resolve) => {
    server = api.listen(port, resolve);
  });
});

afterAll(async () => {
  await new Promise(resolve => server.close(resolve));
});

describe('Feathers application tests', () => {
  it('should have no default route', async () => {
    const actual = await (await fetch(getUrl('/'))).json();
    expect(actual).toMatchSnapshot();
  });
});
