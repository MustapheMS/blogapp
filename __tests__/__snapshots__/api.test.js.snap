// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feathers application tests should have no default route 1`] = `
Object {
  "className": "not-found",
  "code": 404,
  "data": Object {
    "url": "/",
  },
  "errors": Object {},
  "message": "Page not found",
  "name": "NotFound",
}
`;
